docs/langref/relay_type.rst

```diff
@@ -0,0 +1,222 @@
+===================
+Relay's Type System
+===================
+
+We have briefly introduced types while detailing the the expression language
+of Relay, but have not yet described the type system. Relay is
+a statically typed and type-inferred language, allowing programs to
```

Comment:
```suggestion
a statically-typed, type-inferred language, allowing programs to
```

Reply:
Hyphens are usually not recommended after adverbs ending in "-ly."